comma/model.py

```diff
@@ -0,0 +1,190 @@
+"""Model class definition
+"""
+from comma.individual import Individual
+import pandas as pd
+from . import read_json_as_dict, _get_one_hot_encoded_features, PARAMS_INDIVIDUAL, PARAMS_MODEL
+import os
+import numpy as np
+from tqdm import tqdm
+
+
+class Model:
+    _status = dict()
+    _required_params = ['size', 'steps', 'actions', 'status',
+                        'lockdown_policies', 'lockdown']
+
+    def __init__(self):
+        self.agents: list = None
+        self.simulation_id: int = None
+        self.current_step: int = 0 # keep track of the current simulation step
+        self.lockdown_status: dict = {}
+
+    def setup(self, size: int, dir_params: str, use_ipf: bool = False):
```

Comment:
The `__init__()` method is usually the place to set up the object; is there a reason why this is done in the separate `setup()` method here?